### ISSUE TITLE
Retrieve embedded images

### ISSUE THREAD

**Issue by nemiq** [omniphx/forrest#188]
Hi,

Sometimes there are embedded images in the rich content fields. Those images are stored at:
https://c.csXX.content.force.com/servlet/rtaImage

It would be nice if it was possible to retrieve them as described here: https://salesforce.stackexchange.com/questions/44799/how-to-access-servlet-rtaimage-resources-over-the-api

Problem is I can't retrieve the current access token. Custom feature don't accept any other url than normal + /services/apexrest always added.

Something like this would be nice to have:

```
$url = 'https://c.csXX.content.force.com';
$options['parameters'] = [
            'eid' => 'xxxx000000xxx',
            'feoid' => 'xxxx000000xxxx',
            'refid' => 'xxxx00000xxxx'
];
$options['headers'] = [
            'Accept' => 'image/png',
            'Content-Type' => 'image/png',
];
public function retrieveEmbeddedImages($url, $options = [])
    {
        $url .= '/servlet/rtaImage';
        $parameters = [];
        if (isset($options['parameters'])) {
            $parameters = array_replace_recursive($parameters, $options['parameters']);
            $url .= '?'.http_build_query($parameters);
        }
        return $this->request($url, $options);
    } 
```

And request function to be modified to handle other header settings.

**Comment by omniphx** [comment]
Did you try using one of the get, post, put, delete methods? It's not documented that well but you can use any of the public methods in `Client.php`.

If the instance URL is different as it is with the LiveAgent Rest API it might be a problem.

**Comment by nemiq** [comment]
I've given up on the embedded pictures for now, that is a messy part of Salesforce.. However, it is quite difficult to retrieve normal attachments as well. Mainly because handleRequest() enforces JSON encoding regardless of format.

Right now I have modified Facade/Service provider/UserPassword/JSON formatter, which still use JSON formatter but with the option to supply format "custom", then no JSON encoding is applied (based on run time decision). I also added optional headers[] which are used in custom format. In that way I can set correct Accept/Content-type.

Basically I want JSON response except for attachments.
